src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.AllLowerCamelCaseMatcher.cs

```diff
@@ -102,13 +110,14 @@ public AllLowerCamelCaseMatcher(string candidate, bool includeMatchedSpans, Stri
                             contiguous = true;
                         }
 
-                        var (weight, matchedSpans) = TryConsumePatternOrMatchNextHump(
+                        var (weight, matchedSpansInReverse) = TryConsumePatternOrMatchNextHump(
                             patternIndex, humpIndex, contiguous.Value);
                         if (weight == null)
                         {
+                            Debug.Assert(matchedSpansInReverse == null);
```

Comment:
:memo: During my first-pass read, my instinct based on the variable declaration form used above was this check would always fail. It turns out this is not the case, but it goes to show why the implications of `default(T)` are an important factor in a decision to use it.